src/test/clojure/clara/test_java_facts.clj

```diff
@@ -0,0 +1,29 @@
+(ns clara.test-java-facts
+  (:require [clara.tools.testing-utils :as tu]
+            [clara.rules :as rules]
+            [clojure.test :refer [is deftest run-tests testing use-fixtures]])
+  (:import [clara.test.facts
+            BeanTestFact]))
+
+;; A test to demonstrate that a Pojo with indexed property accessors can be used as an alpha root in a session,
+;; see https://github.com/cerner/clara-rules/issues/446
+(tu/def-rules-test test-basic-rule
+  {:rules [kansas-rule [[[BeanTestFact
+                          (= ?locs locations)
+                          (some #(= "Kansas" %) ?locs)]]
+                        (rules/insert! "Kansas Exists")]]
+   :queries [string-query [[] [[?s <- String]]]]
+
+   :sessions [empty-session [kansas-rule string-query] {}]}
+
+  (let [locs (make-array String 2)]
```

Comment:
You could use `doto` around `make-array` to do the `aset`'s all in one go and make it seem somewhat more "functional/immutable" in style

Reply:
done:
https://github.com/cerner/clara-rules/pull/449/commits/1a120651ad7f82c4ca32b6b46a0f60aaf84f94b0

Reply:
+1